Modules/xASL_module_ASL.m

```diff
@@ -183,34 +136,42 @@
 	xASL_adm_SaveX(x);
 end
 
-if ~isfield(x.modules.asl,'motionCorrection')
-    x.modules.asl.motionCorrection = 1;
-end
 
-% Backward compatibility
-if xASL_exist(x.P.Pop_Path_qCBF_untreated,'file')
-    xASL_Move(x.P.Pop_Path_qCBF_untreated, x.P.Pop_Path_qCBF, true);
-end
-
-%% If ASL_4D_parms.mat (ASL parameters) file exist, load and overwrite existing parameters in the xASL file (inheritance principle)
-if ~x.mutex.HasState('999_ready')
-    bO = true; % generate output, some processing has and some has not been yet done
-else
-    bO = false; % skip output, as all processing has been performed
-end
 
+%% D1. Load ASL parameters (inheritance principle)
 [~, x] = xASL_adm_LoadParms(x.P.Path_ASL4D_parms_mat, x, bO);
 
 if ~isfield(x,'Q')
```

Comment:
You can remove the Q-checking here (line 144) as it is already on line 197 at step E1.

Reply:
And `xASL_init_SubStructs` should always add it for your 😄

Reply:
So that means - delete on both places :D

Reply:
Fixed.